Examples/Cxx/Circuits/SP_ReducedOrderSG_VBR_Load_Fault.cpp

```diff
@@ -0,0 +1,166 @@
+						GridParams.VnomMV * sin(GridParams.initVoltAngle)));
+
+	auto load = CPS::SP::Ph1::Load::make("Load", logLevel);
+	load->setParameters(GridParams.initActivePower, GridParams.initReactivePower, 
+						GridParams.VnomMV);
+	
+	//Breaker
+	auto fault = CPS::SP::Ph1::Switch::make("Br_fault", logLevel);
+	fault->setParameters(switchOpen, switchClosed);
+	fault->open();
+
+	// Topology
+	genSP->connect({ n1SP });
+	load->connect({ n1SP });
+	fault->connect({SP::SimNode::GND, n1SP});
+	SystemTopology systemSP;
+	if (SGModel==3)
+		systemSP = SystemTopology(GridParams.nomFreq,
+			SystemNodeList{n1SP},
+			SystemComponentList{std::dynamic_pointer_cast<SP::Ph1::SynchronGenerator3OrderVBR>(genSP), 
+								load, fault});
+	else if (SGModel==4)
+		systemSP = SystemTopology(GridParams.nomFreq,
+			SystemNodeList{n1SP},
+			SystemComponentList{std::dynamic_pointer_cast<SP::Ph1::SynchronGenerator4OrderVBR>(genSP), 
+								load, fault});
+	else if (SGModel==6)
+		systemSP = SystemTopology(GridParams.nomFreq,
+			SystemNodeList{n1SP},
+			SystemComponentList{std::dynamic_pointer_cast<SP::Ph1::SynchronGenerator6aOrderVBR>(genSP), 
+								load, fault});
+	else if (SGModel==7)
+		systemSP = SystemTopology(GridParams.nomFreq,
+			SystemNodeList{n1SP},
+			SystemComponentList{std::dynamic_pointer_cast<SP::Ph1::SynchronGenerator6bOrderVBR>(genSP), 
+								load, fault});
+
+	// Logging
+	auto loggerSP = DataLogger::make(simNameSP, true, logDownSampling);
+	loggerSP->addAttribute("v_gen", 	 genSP->attribute("v_intf"));
+    loggerSP->addAttribute("i_gen", 	 genSP->attribute("i_intf"));
+    loggerSP->addAttribute("Etorque", 	 genSP->attribute("Etorque"));
+    //loggerSP->addAttribute("delta", 	 genSP->attribute("delta"));
+    //loggerSP->addAttribute("w_r", 		 genSP->attribute("w_r"));
+	//loggerSP->addAttribute("Edq0", 		 genSP->attribute("Edq0_t"));
+	//loggerSP->addAttribute("Vdq0", 		 genSP->attribute("Vdq0"));
+	//loggerSP->addAttribute("Idq0", 		 genSP->attribute("Idq0"));
+	//loggerSP->addAttribute("Evbr", 		 genSP->attribute("Evbr"));
+
+	Simulation simSP(simNameSP, logLevel);
+	simSP.doInitFromNodesAndTerminals(true);
+	simSP.setSystem(systemSP);
+	simSP.setTimeStep(timeStep);
+	simSP.setFinalTime(finalTime);
+	simSP.setDomain(Domain::SP);
+	simSP.setMnaSolverImplementation(DPsim::MnaSolverFactory::EigenSparse);
+	simSP.addLogger(loggerSP);
+	simSP.doSystemMatrixRecomputation(true);
+
+	// Events
+	auto sw1 = SwitchEvent::make(startTimeFault, fault, true);
+	simSP.addEvent(sw1);
+
+	auto sw2 = SwitchEvent::make(endTimeFault, fault, false);
+	simSP.addEvent(sw2);
+	
+	simSP.run();
+}
+
+
+int main(int argc, char* argv[]) {	
+
+	// Simulation parameters
+	Real SwitchClosed = GridParams.SwitchClosed;
+	Real SwitchOpen = GridParams.SwitchOpen;
+	Real startTimeFault = 1.0;
+	Real endTimeFault   = 1.1;
+	Real finalTime = 5;
+	Real timeStep = 1e-3;
+	Real H = syngenKundur.H;
+	int SGModel = 4;
```

Comment:
I think it's better that `SGModel` is of type `String`. Then you can use `'6a'` and `'6b'` instead of `6` and `7`, which can become confusing in particular if we have models `6c`, `6d` etc. in the future

Reply:
I agree but it is not possible to use strings with the option -o of the command line. Any idea how I can do it?

Reply:
This will be possible after I have resolved #77

Reply:
#77 has been resolved and merged into this branch. It can be implemented now using
https://github.com/sogno-platform/dpsim/blob/02babaea27ffc515c35b3b6404d719954c7b10cb/Include/dpsim/Utils.h#L146